types/sarif/index.d.ts

```diff
@@ -0,0 +1,1607 @@
+     */
+    parentKey?: string;
+
+    /**
+     * Key/value pairs that provide additional information about the logical location.
+     */
+    properties?: PropertyBag;
+}
+
+/**
+ * Encapsulates a message intended to be read by the end user.
+ */
+export interface Message {
+    /**
+     * An array of strings to substitute into the message string.
+     */
+    arguments?: string[];
+
+    /**
+     * The resource id for a plain text message string.
+     */
+    messageId?: string;
+
+    /**
+     * The resource id for a rich text message string.
+     */
+    richMessageId?: string;
+
+    /**
+     * A rich text message string.
+     */
+    richText?: string;
+
+    /**
+     * A plain text message string.
+     */
+    text?: string;
+
+    /**
+     * Key/value pairs that provide additional information about the message.
+     */
+    properties?: PropertyBag;
+}
+
+/**
+ * Represents a node in a graph.
+ */
+export interface Node {
+    /**
+     * Array of child nodes.
+     */
+    children?: Node[];
+
+    /**
+     * A string that uniquely identifies the node within its graph.
+     */
+    id: string;
+
+    /**
+     * A short description of the node.
+     */
+    label?: Message;
+
+    /**
+     * A code location associated with the node.
+     */
+    location?: Location;
+
+    /**
+     * Key/value pairs that provide additional information about the node.
+     */
+    properties?: PropertyBag;
+}
+
+/**
+ * Describes a condition relevant to the tool itself, as opposed to being relevant to a target being analyzed by the
+ * tool.
+ */
+export interface Notification {
+    /**
+     * The runtime exception, if any, relevant to this notification.
+     */
+    exception?: Exception;
+
+    /**
+     * An identifier for the condition that was encountered.
+     */
+    id?: string;
+
+    /**
+     * A value specifying the severity level of the notification.
+     */
+    level?: Notification.level;
+
+    /**
+     * A message that describes the condition that was encountered.
+     */
+    message: Message;
+
+    /**
+     * The file and region relevant to this notification.
+     */
+    physicalLocation?: PhysicalLocation;
+
+    /**
+     * The stable, unique identifier of the rule (if any) to which this notification is relevant. This member can be
+     * used to retrieve rule metadata from the rules dictionary, if it exists.
+     */
+    ruleId?: string;
+
+    /**
+     * The thread identifier of the code that generated the notification.
+     */
+    threadId?: number;
+
+    /**
+     * The Coordinated Universal Time (UTC) date and time at which the analysis tool generated the notification.
+     */
+    timeUtc?: string;
+
+    /**
+     * Key/value pairs that provide additional information about the notification.
+     */
+    properties?: PropertyBag;
+}
+
+export namespace Notification {
+    // tslint:disable-next-line no-const-enum
```

Comment:
See previous comment

Reply:
Thanks for the feedback, I've updated these to be "type" instead of "const enum"